proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -0,0 +1,1232 @@
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// `unsafeRawPointer` must be correctly aligned for accessing
+  /// an element of type `Element`.
+  ///
+  /// - Parameters:
+  ///   - unsafeRawPointer: a pointer to the first initialized element.
+  ///   - type: the type to use when interpreting the bytes in memory.
+  ///   - count: the number of initialized elements in the view.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  public init<Owner>(
+    unsafeStart pointer: UnsafeRawPointer,
+    byteCount: Int,
+    owner: borrowing Owner
+  ) -> dependsOn(owner) Self
+}
+```
+
+#####  `Collection`-like API:
+
+The following typealiases, properties, functions and subscripts have direct counterparts in the `Collection` protocol hierarchy. Their semantics shall be as described where they counterpart is declared (in `Collection` or `RandomAccessCollection`). The only difference with their counterpart should be a lifetime dependency annotation where applicable, allowing them to return borrowed nonescapable values or borrowed noncopyable values.
+
+```swift
+extension Span {
+  public var count: Int { get }
+  public var isEmpty: Bool { get }
+  public var indices: Range<Int> { get }
+
+  public subscript(_ position: Int) -> dependsOn(self) Element { get }
+}
+```
+
+##### Accessing subranges of elements:
+
+In SE-0437, `UnsafeBufferPointer`'s slicing subscript was replaced by the `extracting(_ bounds:)` functions, due to the copyability assumption baked into the standard library's `Slice` type. `Span` has similar requirements as `UnsafeBufferPointer`, in that it must be possible to obtain an instance representing a subrange of the same memory as another `Span`. We therefore follow in the footsteps of SE-0437, and add a family of `extracting()` functions:
+
+```swift
+extension Span {
+  func extracting(_ bounds: Range<Int>) -> dependsOn(self) Self
+  func extracting(_ bounds: some RangeExpression<Int>) -> dependsOn(self) Self
+  func extracting(_: UnboundedRange) -> dependsOn(self) Self
+}
+```
+
+##### Unchecked access to elements and subranges of elements:
+
+The `subscript` and the `extracting()` functions mentioned above all have always-on bounds checking of their parameters, in order to prevent out-of-bounds accesses. We also want to provide unchecked variants as an alternative for cases where bounds-checking is proving costly.
+
+```swift
+extension Span {
+  // Unchecked subscripting and extraction
+
+  /// Accesses the element at the specified `position`.
+  ///
+  /// This subscript does not validate `position`; this is an unsafe operation.
+  ///
+  /// - Parameter position: The position of the element to access. `position`
+  ///     must be a valid index that is not equal to the `endIndex` property.
+  public subscript(
+    unchecked position: Index
+  ) -> dependsOn(self) Element { get }
```

Comment:
Same question about `dependsOn(self)` and `get`.

Reply:
I audited the `dependsOn` annotations, and it turns out none of them were needed (so they're removed.)
@tbkka believes we should actually mention `_read` accessors, so I will.